File: crs-development/src/crs-api-client/src/main/IEDR-API/bus/commands/CreditCard.php
<?php

/*
 * Copyright (C) 2007 NASK
 * http://www.nask.pl
 * http://www.dns.pl
 */

class CreditCard {

    static public $PAY_CARD_TYPES = array("VISA", "MC", "LASER");
    private $cardHolder;
    private $cardNumber;
    private $expiryDate;
    private $cardType;
    private $cvnNumber;
    private $cvnPresInd;

    function getCardHolder() {
        return $this->cardHolder;
    }

    function setCardHolder($cardHolder) {
        $this->cardHolder = $cardHolder;
    }

    function getCardNumber() {
        return $this->cardNumber;
    }

    function setCardNumber($cardNumber) {
        $this->cardNumber = $cardNumber;
    }

    function getExpiryDate() {
        return $this->expiryDate;
    }

    function setExpiryDate($expiryDate) {
        $this->expiryDate = $expiryDate;
    }

    function getCardType() {
        return $this->cardType;
    }

    function setCardType($cardType) {
        $this->cardType = $cardType;
    }

    function getCvnNumber() {
        return $this->cvnNumber;
    }

    function setCvnNumber($cvnNumber) {
        $this->cvnNumber = $cvnNumber;
        $this->cvnPresInd = 1;
    }

    function getCvnPresInd() {
        return $this->cvnPresInd;
    }

}

File: crs-development/src/crs-core/src/main/java/pl/nask/crs/payment/CreditCard.java
package pl.nask.crs.payment;

public class CreditCard {

    private final String cardNumber;
    private final String cardExpDate;
    private final String cardType;
    private final String cardHolderName;
    private final String cvnNumber;
    private final Integer cvnPresenceIndicator;

    public CreditCard(String cardNumber, String cardExpDate, String cardType, String cardHolderName, String cvnNumber,
            Integer cvnPresenceIndicator) {
        this.cardNumber = normalizedCardNumber(cardNumber);
        this.cardExpDate = cardExpDate;
        this.cardType = cardType;
        this.cardHolderName = cardHolderName;
        this.cvnNumber = cvnNumber;
        this.cvnPresenceIndicator = cvnPresenceIndicator;
    }

    private String normalizedCardNumber(String cardNumber) {
        return cardNumber.replaceAll("\\s", "");
    }

    public String getCardNumber() {
        return cardNumber;
    }

    public String getCardExpDate() {
        return cardExpDate;
    }

    public String getCardType() {
        return cardType;
    }

    public String getCardHolderName() {
        return cardHolderName;
    }

    public String getCvnNumber() {
        return cvnNumber;
    }

    public Integer getCvnPresenceIndicator() {
        return cvnPresenceIndicator;
    }

}
